mesa/space.py

```diff
@@ -1082,4 +1080,8 @@ def iter_cell_list_contents(self, cell_list: list[int]) -> Iterator[Agent]:
         """Returns an iterator of the agents contained in the nodes identified
         in `cell_list`; nodes with empty content are excluded.
         """
-        yield from self.get_cell_list_contents(cell_list)
+        return itertools.chain.from_iterable(
+            self.G.nodes[node_id]["agent"]
+            for node_id in cell_list
+            if not self.is_cell_empty(node_id)
```

Comment:
Is there a particular reason why this is not `filterfalse`?

Reply:
slightly better in terms of performance because nodes are not passed differently in `self.is_cell_empty` and when accessing the agents in them (while in the other classes this happens)

Reply:
Simplicity takes precedence if the perf benefit is small. Let's keep the implementation consistent instead. And I need a up-to-date list of the perf improvement diff relative to main branch to be included in the commit message. I suppose you only need to re-measure for the network grid.

Reply:
Implemented with filterfalse. I recalculated the perf improvement and they are actually smaller than what It seemed to be before, probably something went wrong in the previous perf tests: 1.3x for MultiGrid and more or less the same as before for NetworkGrid (but now the iterator is a real iterator, before it wasn't). The filter functionality will make all of them much better, maybe I will try to implement it in another PR :-)